sdks/java/expansion-service/src/main/java/org/apache/beam/sdk/expansion/service/ExpansionService.java

```diff
@@ -486,6 +515,19 @@ default InputT createInput(Pipeline p, Map<String, PCollection<?>> inputs) {
         .build();
   }
 
+  protected Pipeline createPipeline() {
+    PipelineOptions effectiveOpts = PipelineOptionsFactory.create();
```

Comment:
I created new PipelineOptions per expansion and copy the important options from the command-line. On the other hand I think, that from user perspective this has very low difference, because the main issue here is that the user (with good intentions) specifies an option that is not supported by the service (e.g. most obviously the `runner`). The command line will accept it, but it will be simply silently ignored. I think that is the case we need to address by the validation scope annotations.

Reply:
I think this is OK for now but let's add a TODO and a Jira to implement proper validation.

Reply:
:+1: